### ISSUE TITLE
The simdjson fallback kernel should skip stage 1

### ISSUE THREAD

**Issue by lemire** [simdjson/simdjson#1413]
Current simdjson has a fallback kernel which emulates a high speed wide SIMD stage 1. This is almost certainly wasteful and we should instead move to single-stage model where we only have a stage 2. Possibly after parsing a element, we could conditionally skip allowable white space.

Tentatively marked as 1.0.

**Comment by lemire** [comment]
@jkeiser This becomes obsolete in my mind if we retire the DOM API.

**Comment by jkeiser** [comment]
True. I think it still has value to On Demand ... and it might be easier to prove out in DOM, where more things are held constant.

**Comment by jkeiser** [comment]
I actually wonder if doing this would be faster than haswell and westmere, too . The frontend already repeats a lot of the work of going through numbers and strings. It's puzzling to me how fast we are given that fact. I've been assuming it has to do with branch misses ... but given stage 2 are already branching for the end of each number, and on each backslash and end quote, it feels like the only branch misses stage 1 avoids for us are whitespace skipping.

I'm probably missing something :)
